From f823ee37ca45a7a78160128c97d08af6da689a3b Mon Sep 17 00:00:00 2001
From: Eric Sauvageau <merlin@asuswrt-merlin.net>
Date: Sun, 20 Sep 2026 20:33:53 -0400
Subject: [PATCH] libovpn: enforce a DH of at least 2048-bit, the minimum
 required by the security level OpenVPN asks from OpenSSL 3.x

(cherry picked from commit fd5aae91df90766aad32a522da49c20aff71d289)
---
 release/src/router/libovpn/openvpn_setup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/release/src/router/libovpn/openvpn_setup.c b/release/src/router/libovpn/openvpn_setup.c
index d8752c35d1..53f192f473 100644
--- a/release/src/router/libovpn/openvpn_setup.c
+++ b/release/src/router/libovpn/openvpn_setup.c
@@ -1444,8 +1444,8 @@ void ovpn_write_dh(ovpn_sconf_t *sconf, int unit) {
 					len = DH_bits(dhparams);
 					OPENSSL_free(dhparams);
 				}
-				if ((len != 0) && (len < 1024))
-					logmessage("openvpn","WARNING: DH for server %d is too weak (%d bit, must be at least 1024 bit). Using a pre-generated 2048-bit PEM.", unit, len);
+				if ((len != 0) && (len < 2048))
+					logmessage("openvpn","WARNING: DH for server %d is too weak (%d bit, must be at least 2048 bit). Using a pre-generated 2048-bit PEM.", unit, len);
 			}
 			fclose(fp);
 		}
-- 
2.25.1

